test/apps/iodemo/io_demo.cc

```diff
@@ -614,15 +798,15 @@ class DemoClient : public P2pDemoCommon {
                 break;
             }
 
-            size_t conn_num = IoDemoRandom::rand(0, conn.size() - 1);
+            size_t conn_num = IoDemoRandom::rand(size_t(0), conn.size() - 1);
```

Comment:
maybe  IoDemoRandom::rand<size_t>(0, conn.size() - 1);

Reply:
why it's better?

Reply:
BTW, why was it needed in first place? compilation failed?

Reply:
yes 0 is int by default and compiler can't match func

Reply:
0ul?

Reply:
not sure if it could help for any compiler, usually `typedef ` means new diff type.
why you don't like size_t(0)? constructor of default type should be zero cost.

Reply:
ok